sstables/sstable_directory.hh

```diff
@@ -63,6 +63,7 @@ public:
         bool enable_dangerous_direct_import_of_cassandra_counters = false;
         bool allow_loading_materialized_view = false;
         bool sort_sstables_according_to_owner = true;
+        std::optional<sstable_open_config> sstable_open_config;
```

Comment:
Since `sstable_open_config` has a reasonable default constructor, I think there is no point in making this `std::optional<>`. The way ti is used also supports this:
```c++
    auto sst = co_await load_sstable(std::move(desc), flags.sstable_open_config.value_or(sstable_open_config{}));
```

If this changes in the future, we can always go switch to `std::optional<>`.

Reply:
Makes sense. Will change. I used optional just for making it explicit the "optionalness" of the config :-)